### DISCUSSION TITLE
Display Driver for a SSD1803A Controller

### DISCUSSION THREAD

**Discussion by WolleKette** [micropython/micropython#10623] [Libraries & Drivers]
I need to use a LED display 4x20 characters with a **SSD1803A controller** for a project. Unfortunately I can't find a driver for the controller. Has anyone ever used something like this in Micropython?

**Comment by robert-hh** [comment]
On a first glance, the interface looks similar to a HD44780. So you could use @dhylands driver at https://github.com/dhylands/python_lcd.

**Reply by WolleKette** [reply]
The SSD1803A controller supports up to 4 lines of 20 characters each while the HD44780U is probably limited to 2 lines. That doesn't sound very similar to me.

**Reply by mcauser** [reply]
The HD44780 supports 4 line / 20 char displays (20x4). Most commonly used with 16x2.
The displays always have the full 80 characters available, but often in these drivers they make them wrap after they move off-screen. On a full 20x4, you don't need any additional code, it automatically wraps back to the beginning.

**Comment by robert-hh** [comment]
I used the driver from the link I gave you to control 4 x 20 sized LCD displays. So you could try it. The command set of both chips looks identical. The only difference between a 2x16 and 4x20 display is the RAM size for the contents.
Edit: Setting the RAM address includes a 7 bit address field. So it can address 80 bytes.

**Reply by mcauser** [reply]
On my 16x2's, they still have the full 80 content bytes in RAM, but they are just off-screen.
eg. if you start at pos 0,0 and write 80x A's then write a B, it'll show a B at pos 0,0 and the rest filled with A.

**Reply by WolleKette** [reply]
Ok. I will test it and report here. But it will take a few days. 
Thanks in any case for now.
